lib/go/edgecontext/validator.go

```diff
@@ -6,17 +6,37 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/golang-jwt/jwt/v4"
 	"github.com/reddit/baseplate.go/secrets"
-	jwt "github.com/reddit/jwt-go/v3"
+	"golang.org/x/crypto/ssh"
 )
 
-type keysType = []*rsa.PublicKey
+type keysType struct {
+	// map of kid -> pub key.
+	m map[string]*rsa.PublicKey
+
+	// when either kid header does not exist in the jwt token,
+	// or the kid is not present in the map,
+	// we fallback to the first (usually current) key.
+	first *rsa.PublicKey
```

Comment:
I would either call it "fallback" (explains how it's used, not how it's computed), or "current" (it's a better "implementation detail" to hard-code as a field name than "first")

Reply:
I disagree that "fallback" is a better name than "first" here as "first" actually provides more info than "fallback".

Basically for a versioned secret, its `GetAll` returns secrets in this order: current, prev, next (this is documented [here](https://pkg.go.dev/github.com/reddit/baseplate.go/secrets#VersionedSecret.GetAll)), so the implementation detail of first vs. for example last actually matters here. actually I'm going to add some tests to make sure that we actually set the first key here.

(I originally called it "current" but changed to "first" because in the rare case that the current secret cannot be parsed as RSA pub key, the first will be a different one and not really current).